File: week-2/sort-characters-by-frequency/sort-characters-by-frequency.js
/**
 * @param {string} s
 * @return {string}
 */
var frequencySort = function(s) {
    const hashtable = {}
    for (let i of s) {
        if (i in hashtable)
            ++hashtable[i];
        else
            hashtable[i] = 1;
    }
    
    const letters = Object.entries(hashtable).sort((a, b) => a[1]-b[1]);
    let res = '';
    for (let i of letters.reverse()) {
        res += new Array(i[1]).fill(i[0]).join('');
    }
    
    return res;
};

File: week-2/sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        hashtable = {}
        for i in s:
            hashtable[i] = hashtable[i] + 1 if i in hashtable else 1
        
        letters = list(hashtable.items())
        letters.sort(key=lambda x: x[1])
        
        res = []
        for i in letters[::-1]:
            res += i[0]*i[1]
        
        return res
        
